sympy/physics/control/lti.py

```diff
@@ -1235,6 +1457,202 @@ def is_biproper(self):
+        >>> G1 = TransferFunction(p**2 + 2*p + 4, p - 6, p)
+        >>> G2 = TransferFunction(p, 4 - p, p)
+        >>> G3 = TransferFunction(0, p**4 - 1, p)
+        >>> Parallel(G1, G2).var
+        p
+        >>> Parallel(-G3, Series(G1, G2)).var
+        p
+
+        """
+        return self.args[0].var
+
+    @property
+    def num_inputs(self):
+        """Returns the number of input signals of the parallel system."""
+        return self.args[0].num_inputs
+
+    @property
+    def num_outputs(self):
+        """Returns the number of output signals of the parallel system."""
+        return self.args[0].num_outputs
+
+    @property
+    def shape(self):
+        """Returns the shape of the equivalent MIMO system."""
+        return self.args[0].num_outputs, self.args[0].num_inputs
+
+    def doit(self, **kwargs):
+        """
+        Returns the resultant transfer function matrix obtained after evaluating
+        the MIMO systems arranged in a parallel configuration.
+
+        Examples
+        ========
+
+        >>> from sympy.abc import s, p, a, b
+        >>> from sympy.physics.control.lti import TransferFunction, MIMOParallel, TransferFunctionMatrix
+        >>> tf1 = TransferFunction(a*p**2 + b*s, s - p, s)
+        >>> tf2 = TransferFunction(s**3 - 2, s**4 + 5*s + 6, s)
+        >>> tfm_1 = TransferFunctionMatrix([[tf1, tf2], [tf2, tf1]])
+        >>> tfm_2 = TransferFunctionMatrix([[tf2, tf1], [tf1, tf2]])
+        >>> MIMOParallel(tfm_1, tfm_2).doit()
+        TransferFunctionMatrix(((TransferFunction((-p + s)*(s**3 - 2) + (a*p**2 + b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s), TransferFunction((-p + s)*(s**3 - 2) + (a*p**2 + b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s)), (TransferFunction((-p + s)*(s**3 - 2) + (a*p**2 + b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s), TransferFunction((-p + s)*(s**3 - 2) + (a*p**2 + b*s)*(s**4 + 5*s + 6), (-p + s)*(s**4 + 5*s + 6), s))))
+
+        """
+        _arg = (arg.doit()._expr_mat for arg in self.args)
+        res = MatAdd(*_arg, evaluate=True)
+        return TransferFunctionMatrix.from_Matrix(res, self.var)
+
+    def _eval_rewrite_as_TransferFunctionMatrix(self, *args, **kwargs):
+        return self.doit()
+
+    def __add__(self, other):
+        if not isinstance(other, LinearTimeInvariant):
+             return NotImplemented
+
+        if other.is_SISO:
+            raise TypeError("Cannot add SISO and MIMO systems together.")
+
+        if isinstance(other, MIMOParallel):
+            self_arg_list = list(self.args)
+            other_arg_list = list(other.args)
+            return MIMOParallel(*self_arg_list, *other_arg_list)
+
+        self_arg_list = list(self.args)
+        return MIMOParallel(*self_arg_list, other)
+
+    __radd__ = __add__
+
+    def __sub__(self, other):
+        if not isinstance(other, MIMOLinearTimeInvariant):
+             return NotImplemented
```

Comment:
Maybe there should be a decorator to do this check and return `NotImplemented` since it is used so many times. Already int he codebase there is `sympifyit` and `polyfyit` that do something similar